compose/ui/ui-test/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skiko.kt

```diff
@@ -205,30 +228,50 @@ open class SkikoComposeUiTest @InternalTestApi constructor(
     private val testOwner = SkikoTestOwner()
     private val testContext = TestContext(testOwner)
 
-    fun <R> runTest(block: SkikoComposeUiTest.() -> R): R {
-        return composeRootRegistry.withRegistry {
-            withScene {
+    fun runTest(
+        block: suspend SkikoComposeUiTest.() -> Unit
+    ): TestResult {
+        composeRootRegistry.setupRegistry()
```

Comment:
Why not instead put `composeRootRegistry.withRegistry { ... }` inside the call to `kotlinx.coroutines.test.runTest`?

Reply:
> The composeRootRegistry should be started before the scene is created.

Maybe I missed that. Why does it need to be started before the scene is created?

Reply:
> Why does it need to be started before the scene is created?
Otherwise the registry won't contain any `PlatformRootForTest` registered. It relies on a callback `onRootForTestCreated`, which is triggered when RootNodeOwner is created (and it's created when along with CanvasLayersComposeSceneImpl).